openmls/src/treesync/node/leaf_node.rs

```diff
@@ -382,6 +384,26 @@ impl TlsDeserializeTrait for LeafNodeTbs {
     }
 }
 
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsDeserialize)]
+pub struct StateUnvalidated;
+
+// TODO: Name up to debate.
+#[derive(Debug, Clone, PartialEq, Eq, TlsSize, TlsSerialize)]
+pub struct StateKeyPackage;
```

Comment:
This leaf node was validated in the context of a key package. The MLS draft lists the required checks. Notably, the leaf node must have a key package leaf node source and a valid lifetime (and lifetime range).